runtime/bf16_support.cu

```diff
@@ -16,6 +16,42 @@ __device__ __inline__ __bfloat __float2bfloat(const float);
 struct __align__(2) __bfloat {
   __bfloat() = default;
 
+  __bfloat(const __bfloat& other) {
+    __x = other.__x;
+  }
+
+  __bfloat(const volatile __bfloat& other) {
+    __x = other.__x;
+  }
+
+  __bfloat(const volatile __bfloat&& other) {
+    __x = other.__x;
+  }
+
+  __device__ void operator=(const __bfloat& other) {
+    __x = other.__x;
+  }
+
+  __device__ void operator=(const volatile __bfloat& other) {
+    __x = other.__x;
+  }
+
+  __device__ void operator=(const volatile __bfloat&& other) {
```

Comment:
non-volatile temporary is just an rvalue and it can be bound to `const T&`. 

i.e. if you have
```
void foo(const T& x);
foo(std::move(x)); // This will call into `void foo(const T& x)`
```

Reply:
Do you think it's better to leave an explicit move here, in case we need to change behavior in the future?

To be fair, @jacobhinkle 's [comment earlier](https://github.com/NVIDIA/Fuser/pull/2269#discussion_r1606698934) regarding the various overload of `volatile` on lhs/rhs operands falls into about the same bucket here. i.e. generated code conceptually isn't any different, but having an explicit overload written out might help to eliminate the confusion and doubts.

Reply:
Added the overload of `const T&&` to avoid overload resolution going through `const volatile T&&` preceding `const T&`. Detailed put in Note 1 in PR description.
Thanks for catching this one.